### PR TITLE
[v1.4] Add word dictionary settings

### LINKED ISSUE #569: [v1.4] [Prototype] User dictionary settings API
**Description**
Introduction of a new setting, `wordDictionary` in Meilisearch 1.4 as a prototype. We want to create a prototype version of `meilisearch-php`, implementing the required changes to be able to try out the new settings. 

- issue: https://github.com/meilisearch/meilisearch/issues/3917
- spec: https://github.com/meilisearch/specifications/pull/255
- discussion: https://github.com/meilisearch/product/discussions/663

**To do**

- [x] GET route that fetches the setting
- [x] PUT route that updates the setting
- [x] DELETE route that resets the setting
- [x] Add tests

**Basic example**

```php
$client->index('indexName')->getWordDictionnary();
$client->index('indexName')->updateWordDictionnary(['x']);
$client->index('indexName')->resetWordDictionary();
```

### PR DESCRIPTION
**Pull request by alallema**
# Pull Request

## Related issue
Fixes #569 

## What does this PR do?
- Add dictionary settings
- Create tests

### CODE REVIEW

File: tests/Settings/WordDictionaryTest.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        parent::setUp();
+        $this->index = $this->createEmptyIndex($this->safeIndexName());
+    }
+
+    public function testGetDefaultWordDictionary(): void
+    {
+        $response = $this->index->getWordDictionary();
+
+        $this->assertEquals(self::DEFAULT_WORD_DICTIONNARY, $response);
```
Comment (norkunas) [inline tests/Settings/WordDictionaryTest.php:26]:
```suggestion
        $this->assertSame(self::DEFAULT_WORD_DICTIONNARY, $response);
```

File: tests/Settings/WordDictionaryTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        ];
+
+        $promise = $this->index->updateWordDictionary($newWordDictionary);
+
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $rankingRules = $this->index->getWordDictionary();
+
+        $this->assertEquals($newWordDictionary, $rankingRules);
```
Comment (norkunas) [inline tests/Settings/WordDictionaryTest.php:43]:
```suggestion
        $this->assertSame($newWordDictionary, $rankingRules);
```

File: tests/Settings/WordDictionaryTest.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public function testResetWordDictionary(): void
+    {
+        $promise = $this->index->resetWordDictionary();
+
+        $this->assertIsValidPromise($promise);
+
+        $this->index->waitForTask($promise['taskUid']);
+        $rankingRules = $this->index->getWordDictionary();
+
+        $this->assertEquals(self::DEFAULT_WORD_DICTIONNARY, $rankingRules);
```
Comment (norkunas) [inline tests/Settings/WordDictionaryTest.php:55]:
```suggestion
        $this->assertSame(self::DEFAULT_WORD_DICTIONNARY, $rankingRules);
```

File: src/Endpoints/Delegates/HandlesSettings.php [hunk 4]
```diff
@@ -275,4 +275,28 @@ public function resetTypoTolerance(): array
...
+     */
+    public function getWordDictionary(): array
+    {
+        return $this->http->get(self::PATH.'/'.$this->uid.'/settings/dictionary');
+    }
+
+    /**
+     * @param list<non-empty-string> $wordDictionary
+     */
+    public function updateWordDictionary(array $wordDictionary): array
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesSettings.php:293]: Can we typehint returned array structure?

File: src/Endpoints/Delegates/HandlesSettings.php [hunk 5]
```diff
@@ -275,4 +275,28 @@ public function resetTypoTolerance(): array
...
+    {
+        return $this->http->put(self::PATH.'/'.$this->uid.'/settings/dictionary', $wordDictionary);
+    }
+
+    public function resetWordDictionary(): array
+    {
+        return $this->http->delete(self::PATH.'/'.$this->uid.'/settings/dictionary');
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesSettings.php:300]: Can we typehint returned array structure?

**Comment by norkunas** [comment]
I'd suggest to update CI to run with the docker image that supports these settings to see if tests works :)

File: src/Endpoints/Delegates/HandlesSettings.php (hunk 4 above)
Comment (alallema) [reply src/Endpoints/Delegates/HandlesSettings.php:293]: Yes, I thought about it, but all setting updates return a task, and none are specified, so I didn't put anything in.

Comment (norkunas) [reply src/Endpoints/Delegates/HandlesSettings.php:294]: Then you could declare the Task return type in some file and then import the type where a task is returned,but then maybe in a separate PR :)

Comment (alallema) [reply src/Endpoints/Delegates/HandlesSettings.php:294]: I will create an issue about it

Comment (norkunas) [reply src/Endpoints/Delegates/HandlesSettings.php:294]: An alternative could be to introduce some value object that would hold the task info

**Comment by alallema** [comment]
> I'd suggest to update CI to run with the docker image that supports these settings to see if tests works :)
The pre-release arrives on Monday and I'll just change the base branch for a bump so that the tests can run on it.

File: tests/Settings/WordDictionaryTest.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+namespace Tests\Settings;
+
+use Meilisearch\Endpoints\Indexes;
+use Tests\TestCase;
+
+final class WordDictionaryTest extends TestCase
+{
+    private Indexes $index;
+
+    public const DEFAULT_WORD_DICTIONNARY = [];
```
Comment (brunoocasali) [inline tests/Settings/WordDictionaryTest.php:14]:
```suggestion
    public const DEFAULT_WORD_DICTIONARY = [];
```

File: tests/Settings/WordDictionaryTest.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    public function testResetWordDictionary(): void
+    {
+        $promise = $this->index->resetWordDictionary();
+
+        $this->assertIsValidPromise($promise);
+
+        $this->index->waitForTask($promise['taskUid']);
+        $wordDictionary = $this->index->getWordDictionary();
+
+        $this->assertSame(self::DEFAULT_WORD_DICTIONNARY, $wordDictionary);
```
Comment (brunoocasali) [inline tests/Settings/WordDictionaryTest.php:55]:
```suggestion
        $this->assertSame(self::DEFAULT_WORD_DICTIONARY, $wordDictionary);
```

File: tests/Settings/WordDictionaryTest.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        $wordDictionary = $this->index->getWordDictionary();
+
+        $this->assertSame($newWordDictionary, $wordDictionary);
+    }
+
+    public function testResetWordDictionary(): void
+    {
+        $promise = $this->index->resetWordDictionary();
+
+        $this->assertIsValidPromise($promise);
```
Comment (brunoocasali) [inline tests/Settings/WordDictionaryTest.php:50]:
```suggestion
```

File: tests/Settings/WordDictionaryTest.php [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    public function testUpdateWordDictionary(): void
+    {
+        $newWordDictionary = [
+            'J. K.',
+            'J. R. R.'
+        ];
+
+        $promise = $this->index->updateWordDictionary($newWordDictionary);
+
+        $this->assertIsValidPromise($promise);
```
Comment (brunoocasali) [inline tests/Settings/WordDictionaryTest.php:38]:
```suggestion
```

**Review by brunoocasali — CHANGES_REQUESTED** [review]

**Review by brunoocasali — APPROVED** [review]
LGTM!

**Pull request merged**